src/Plugin/Notifier/Manager.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\message_notify\Plugin\Notifier;
+
+use Drupal\Component\Plugin\Factory\DefaultFactory;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+use Drupal\message\MessageInterface;
+
+/**
+ * Notifier plugin manager.
+ */
+class Manager extends DefaultPluginManager {
+
+  /**
+   * {@inheritdoc}
+   *
+   * Allow the message entity to be passed to the constructor.
+   */
+  public function createInstance($plugin_id, array $configuration = [], MessageInterface $message = NULL) {
```

Comment:
Prior to this, there was a separate method on the plugins `::init(Message $message)` that was manually called after creating the plugin instance. After talking with Tim Plunkett at Drupalcon, the way to avoid that is to create a custom plugin manager where we controll the `createInstance` method, which allows us to inject the message object to the plugin constructor. See the change [here](https://github.com/Gizra/message_notify/pull/21/files#diff-158232617abd0b1f4a57ee16834c2960L59) for more context.